caddyconfig/httpcaddyfile/addresses.go

```diff
@@ -199,7 +199,7 @@ func (st *ServerType) listenerAddrsForServerBlockKey(sblock serverBlock, key str
 	}
 
 	// the bind directive specifies hosts, but is optional
-	var lnHosts []string
+	lnHosts := make([]string, 0, len(sblock.pile))
```

Comment:
This is not the right length. I don't think we know the right length without an extra iteration. I guess that's OK though since the third parameter is just capacity, not length, and it will never be too large right?

Reply:
Yep. There a couple of them where the length is just the lower bound, others are upper bound, and some are exactly as expected. The point is to avoid constant reallocation by the runtime, and at times avoiding spendy allocation. So there is 1 of 3 scenarios where the capacity is lower than the real number, and we'll endure 1 or 2 more allocations.

Reply:
Alrighty, sounds fine to me. Thanks!